packages/microcosm/src/proto.js

```diff
@@ -0,0 +1,23 @@
+/* @flow */
+
+/**
+ * Given a class and an object, convert a POJO into an ancestor
+ */
+export function inherit(config: *, Parent: Class<*>): * {
+  if (typeof config === 'function') {
+    return config
+  }
+
+  class NewEntity extends Parent {}
+
+  for (var key in config) {
+    Object.defineProperty(NewEntity.prototype, key, {
+      configurable: true,
+      enumerable: false,
+      writeable: true,
+      value: config[key]
+    })
+  }
+
+  return NewEntity
+}
```

Comment:
Does this have a test?

Reply:
It gets covered through domain/effect usage in other tests, but thanks for calling it out. I should test to cover a few cases:

- `null`
- There are certain rules for [property descriptors of ES6 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptors), I should make this play nice

Really, I don't want to have something like this in Microcosm, but we've allowed users to pass domains/effects as POJOs forever. I'm thinking about deprecating it, but actually enforcing that domains/effects need to inherit from a library class might need to go out in 14.x.